### PR TITLE
28-FE-feature-add-home-responsive

### PR DESCRIPTION
**Pull request by treee97**
# Updated Changes
- Now userDetails is only composed of the **main component** and the **content component**. The modal was removed and remade on the main component to allow a more smooth transition between media queries.
- Added context to manage a state in multiple components in media query.

### This PR focuses on adding responsive behaviour.

Because the **user details** and **filters** will be displayed in two different formats, one as a modal for smaller screens and the other as a normal component for bigger screens, I decided to divide both components in 3 pieces:

- Main component
- Content component
- Modal component

The **main** component is the one being displayed in bigger screens. Inside the **content component** can be found.
The **modal** component is the one displayed in smaller screens. The **content component** is also here.
The **content** component is where the data is placed.

This PR also fixed some scroll errors

- Removed flex in the parent elements and used grid
- Added mockup components to userDetails for better understanding while creating the responsive behaviour.

## Feedback
- Buttons of **Save users** and **Message users** in the white board where commented out temporarily. To be discuss.
- userDetailsContent must be made another issue to finish the component and children.
- ~userDetailsModal works like DaisyUI intended to. However in the filtersModal I had to use a local boolean state to close and open it.~ 

image
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
-    "dev": "vite",
```
Comment (vicenlu) [inline package.json:7]: Se escapa al scope del PR no?

File: src/components/ProfileCard.tsx [hunk 2]
```diff
@@ -17,33 +17,47 @@ const ProfileCard: React.FC<Profile> = ({
 }) => {
   const [isChecked, setIsChecked] = useState(false);
   const dispatch = useAppDispatch();
-  const handleUserDetail = () => {
+  const isMobile = window.innerWidth <= 768;
+
+  useEffect(() => {
+    if (isMobile && document) {
+      const userDetailModal = document.getElementById(
```
Comment (vicenlu) [inline src/components/ProfileCard.tsx:24]: Tenemos que comentar por qué has usado `getElementById`, intentaría evitar usarlo, está "fuera" de react (no se suele usar, solo como workaround).
Buscaría una herramienta que nos proporcione React o gestionarlo con un state.

File: src/components/ProfileCard.tsx [hunk 3]
```diff
@@ -17,33 +17,47 @@ const ProfileCard: React.FC<Profile> = ({
...
-    >
+    <div className="mb-4 mr-8 flex w-full cursor-pointer rounded-2xl p-3 hover:bg-[#F2F2F2] md:max-w-[380px]">
       <div className="form-control flex items-center">
         <label className="label cursor-pointer">
           <input
             type="checkbox"
             checked={isChecked}
             onChange={() => setIsChecked(!isChecked)}
-            className="checkbox h-6 w-6 rounded-sm border-2 border-gray-500"
+            className="checkbox h-6 w-6 rounded-md border-2 border-gray-500 [--chkbg:theme(colors.pink.700)] [--chkfg:white] checked:border-transparent"
```
Comment (vicenlu) [inline src/components/ProfileCard.tsx:46]: Mejor crear un tema de daisyUI, no es operativo tener que meter estas clases cuando creemos cada checkbox. Además puede ser fuente de bugs visuales ya que alguien se puede olvidar estas clases.

File: src/components/filters/FiltersContent.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          <input type="checkbox" className="checkbox mr-2" />
+          <span>Frontend</span>
+        </label>
+        <label className="label cursor-pointer justify-start p-1">
+          <input type="checkbox" className="checkbox  mr-2" />
+          <span>Backend</span>
+        </label>
+        <label className="label cursor-pointer justify-start p-1">
+          <input type="checkbox" className="checkbox  mr-2" />
+          <span>Fullstack</span>
```
Comment (vicenlu) [inline src/components/filters/FiltersContent.tsx:17]: Hay mucho código repetido, ya miraremos como renderizar los checkboxes con un loop

File: src/components/ProfileCard.tsx (hunk 3 above)
Comment (treee97) [reply src/components/ProfileCard.tsx]: Con la próxima PR del tema en tailwind esto se solucionará. Aunque las variables creo que son necesarias para poder editar el color del checkbox. Está por verse 🤔

File: src/components/filters/FiltersContent.tsx (hunk 4 above)
Comment (treee97) [reply src/components/filters/FiltersContent.tsx:17]: Como era el código de mi compañera no lo he tocado mucho pero se podría hacer un componente y crear un array con objetos para cada filtro y mapear. Nuevo issue!

**Review by Dar-ya218 — COMMENTED** [review]
Great job done!

**Review by Dar-ya218 — APPROVED** [review]

**Pull request merged**